Updated information about availibility of C# specs

svick opened the pull request:
VS 2013 should also contain the 5.0 spec, but I expect more people have VS 2015 installed, so I think it makes sense to update this.

Also, I don't know how relevant VS Express is and whether it would make sense to remove the note about it.

cc: @BillWagner

svick committed:
Updated information about availability of C# specs

mairaw commented on docs/csharp/language-reference/language-specification.md:
```diff
@@ -34,8 +34,10 @@ translation.priority.ht:
 The C# Language Specification is the definitive source for C# syntax and usage. This spec contains detailed information about all aspects of the language, including many points that the documentation for Visual C# doesn't cover.  
   
- You can download this spec from the [Microsoft Download Center](http://www.microsoft.com/download/details.aspx?id=7029). If you've installed Visual Studio 2013, you can also find the spec on your computer in the Program Files (x86)/Microsoft Visual Studio 12.0/VC#/Specifications/1033 folder. However, installations of Visual Studio Express 2013 don't include this file.  
+You can download version 5.0 of this spec from the [Microsoft Download Center](http://www.microsoft.com/download/details.aspx?id=7029). If you've installed Visual Studio 2015, you can also find the spec on your computer in the Program Files (x86)/Microsoft Visual Studio 14.0/VC#/Specifications/1033 folder. However, installations of Visual Studio Express 2015 don't include this file.
```
given that we don't recommend people to use Express anymore and instead use the Community edition (see https://www.visualstudio.com/vs/visual-studio-express/), I don't think this information about Express is needed any longer (and Express is not even listed in the editions comparison on the VS site). 

Also, I'm wondering if the spec will always be in 1033 folder. 1033 folder is the locale for en-us. If I install VS in Brazilian Portuguese, would the spec be somewhere else? I can test this on a VM.

BillWagner replied:
@mairaw We should check that. I don't believe the word doc was ever localized. @madstorgersen would know for sure.

mairaw replied:
I finally managed to verify this. I've installed VS 2015 in Italian in a VM. The path definitely varies by locale. The spec was under C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC#\Specifications\1040

But the most surprising thing to me was to learn that the spec was also translated:
![image](https://cloud.githubusercontent.com/assets/12971179/23526967/e0a519da-ff48-11e6-9487-24c795b878a4.png)

mairaw commented on docs/csharp/language-reference/language-specification.md:
```diff
@@ -34,8 +34,10 @@ translation.priority.ht:
- You can download this spec from the [Microsoft Download Center](http://www.microsoft.com/download/details.aspx?id=7029). If you've installed Visual Studio 2013, you can also find the spec on your computer in the Program Files (x86)/Microsoft Visual Studio 12.0/VC#/Specifications/1033 folder. However, installations of Visual Studio Express 2013 don't include this file.  
+You can download version 5.0 of this spec from the [Microsoft Download Center](http://www.microsoft.com/download/details.aspx?id=7029). If you've installed Visual Studio 2015, you can also find the spec on your computer in the Program Files (x86)/Microsoft Visual Studio 14.0/VC#/Specifications/1033 folder. However, installations of Visual Studio Express 2015 don't include this file.
+
+Version 6.0 of the spec is not currently available.
```
nit: I was checking our internal style guide and we should never user just spec, always spell out specification. So this article wasn't following the guidelines originally.

mairaw reviewed (commented):
Updating to more recent versions makes sense to me. I'm just wondering if we can change the text in a way that we teach people how to find it, instead of just telling the VS 2015 path. Perhaps if we say that the VS folder name will change depending on the version you have installed?

svick committed:
Addressed feedback

svick committed:
Added information about C# 6 spec draft

svick commented:
Sorry for the huge delay. I have addressed the feedback.

For the path, I don't want to bog down this document with details on how to construct it, so I just added a sentence that basically says "figure it out". Do you think that is sufficient?

I have also added a link to the "unofficial draft" of the C# 6.0 spec in the csharplang repo.

BillWagner reviewed (approved):
Thanks @svick for all your contributions. We appreciate it.

I'll merge this now.
